Assets/Scripts/GUI/MultiplayerPanel.cs

```diff
@@ -0,0 +1,314 @@
+            if (MultiplayerManager.m_Instance.State == ConnectionState.INITIALIZED || MultiplayerManager.m_Instance.State == ConnectionState.DISCONNECTED)
+            {
+                MultiplayerManager.m_Instance.Connect();
+            }
+        }
+
+        protected override void OnDisablePanel()
+        {
+            base.OnDisablePanel();
+
+            if (MultiplayerManager.m_Instance == null) return;
+            if (MultiplayerManager.m_Instance.State != ConnectionState.IN_ROOM)
+            {
+                MultiplayerManager.m_Instance.Disconnect();
+            }
+        }
+
+        private static string GenerateUniqueRoomName()
+        {
+            const int maxAttempts = 10;
+            string roomName;
+            int attempts = 0;
+
+            do
+            {
+                roomName = GenerateRandomRoomName();
+                attempts++;
+            } while (MultiplayerManager.m_Instance != null &&
+             MultiplayerManager.m_Instance.DoesRoomNameExist(roomName) &&
+             attempts < maxAttempts);
+
+            if (attempts >= maxAttempts)
+            {
+                return "default room";
+            }
+
+            return roomName;
+        }
+
+        private static string GenerateRandomRoomName()
+        {
+            System.Random random = new System.Random();
+            return random.Next(100000, 999999).ToString();
+        }
+
+        private void UpdateDisplay()
+        {
+            if (m_RoomNumber) m_RoomNumber.text = "RoomName: " + data.roomName;
+            if (m_Nickname) m_Nickname.text = "Nickname: " + NickName;
+            Alerts();
+        }
+
+        private async void Connect()
+        {
+            if (MultiplayerManager.m_Instance != null)
+            {
+                await MultiplayerManager.m_Instance.Connect();
+            }
+        }
+
+        private async void JoinRoom()
+        {
+
+            if (MultiplayerManager.m_Instance != null)
+            {
+                await MultiplayerManager.m_Instance.JoinRoom(data);
+            }
+        }
+
+        private async void LeaveRoom()
+        {
+            if (MultiplayerManager.m_Instance != null)
+            {
+                await MultiplayerManager.m_Instance.LeaveRoom(false);
+            }
+        }
+
+        private async void Disconnect()
+        {
+            if (MultiplayerManager.m_Instance != null)
+            {
+                await MultiplayerManager.m_Instance.Disconnect();
+            }
+        }
+
+        private void OnStateUpdated(ConnectionState newState)
+        {
+            if (!m_State) return;
+            m_State.text = "State: " + StateToString(newState);
+            UpdateDisplay();
+        }
+
+        private string StateToString(ConnectionState newState)
+        {
+            switch (newState)
+            {
+                case ConnectionState.INITIALISING:
+                    return "Initialising";
```

Comment:
Either make this INITIALIZING with a Z, or change the next one to INITIALISED with an S. (American vs British spelling)

Reply:
went for Z